### PR TITLE
Add EventCounters to AotCompatibility.Test

### PR DESCRIPTION
**Pull request by eerhardt**
This ensures that the OpenTelemetry.Instrumentation.EventCounters library stays AOT-compatible.

Contributes to https://github.com/open-telemetry/opentelemetry-dotnet/issues/3429

cc @Yun-Ting

### CODE REVIEW

**Comment by utpilla** [comment]
@eerhardt The AOT workflow does not always run. You would have to change the path here: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/.github/workflows/ci-aot.yml#L6-L8 and also just test the AOT compatibility outside of the workflow to ensure that EventCounters instrumentation is actually AOT-compliant.

**Comment by Yun-Ting** [comment]
@eerhardt, please take this commit: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1324/commits 
(I can't push to your branch.)
@utpilla: thanks for reviewing! `OpenTelemetry.Instrumentation.EventCounters` is currently AOT-compliant. See: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/actions/runs/5978918066/job/16221944116?pr=1324

**Review by Yun-Ting — APPROVED** [review]
LGTM assuming GitHub workflow file got updated. Thanks!

**Comment by eerhardt** [comment]
> @eerhardt The AOT workflow does not always run. You would have to change the path here: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/.github/workflows/ci-aot.yml#L6-L8 and also just test the AOT compatibility outside of the workflow to ensure that EventCounters instrumentation is actually AOT-compliant.

I've tested it locally, and it worked. Is there a way to manually kick off the CI leg on this change?

@Yun-Ting's Draft PR had these same changes, and the AOT leg passed on it:  https://github.com/open-telemetry/opentelemetry-dotnet-contrib/actions/runs/5978918066/job/16221944116

File: .github/workflows/ci-aot.yml [hunk 1]
```diff
@@ -6,6 +6,8 @@ on:
     paths:
     - 'src/OpenTelemetry.Instrumentation.Runtime/**'
     - '!src/OpenTelemetry.Instrumentation.Runtime/README.md'
+    - 'src/OpenTelemetry.Instrumentation.EventCounters/**'
+    - '!src/OpenTelemetry.Instrumentation.EventCounters/README.md'
   pull_request:
```
Comment (utpilla) [inline .github/workflows/ci-aot.yml:11]: Add the path for the `pull_request` event as well.

Comment (eerhardt) [reply .github/workflows/ci-aot.yml:11]: Good call. Fixed.

**Comment by utpilla** [comment]
> Is there a way to manually kick off the CI leg on this change?

@eerhardt You could add a simple `workflow_dispatch` event to the workflow and then you should be able to manually run the CI on your fork: https://docs.github.com/en/actions/using-workflows/manually-running-a-workflow.

**Comment by eerhardt** [comment]
Thanks for the help! I was able to successfully run the workflow in my fork:

https://github.com/eerhardt/opentelemetry-dotnet-contrib/actions/runs/5979901435/job/16224947116

**Review by utpilla — APPROVED** [review]

**Pull request merged**